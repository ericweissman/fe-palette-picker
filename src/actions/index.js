export const isLoading = (bool) => ({
  type: 'IS_LOADING',
  isLoading: bool
});

export const hasErrored = (message) => ({
  type: 'HAS_ERRORED',
  message
});

//projects
export const getProjectsSuccess = (projects) => ({
  type: 'GET_PROJECTS_SUCCESS',
  projects
})

export const addProjectSuccess = (project) => ({
  type: 'ADD_PROJECT_SUCCESS',
  project
})