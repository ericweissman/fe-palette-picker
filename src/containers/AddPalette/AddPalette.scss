.add-palette {
  @extend .create-project;
  margin: 10px;
  padding: 10px;
  text-align: center;

  select {
    height: 30px;
    background-color: $main-background;
    color: white;
    outline: none;
    font-size: 16px;
    width: 200px;
    border: none;
  }
  button {
    margin-right: 20px;
  }
}