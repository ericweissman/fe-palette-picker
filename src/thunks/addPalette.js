import { isLoading, hasErrored, addPaletteSuccess } from '../actions'

export const addPalette = (url, palette) => {
  return async (dispatch) => {
    try {
      dispatch(isLoading(true))
      const response = await fetch(url, {
        method: "POST",
        headers: {
          "Content-Type": "application/json"
        },
        body: JSON.stringify(palette)
      })
      if (!response.ok) {
        throw Error(response.statusText)
      }
      dispatch(isLoading(false))
      const result = await response.json()
      dispatch(addPaletteSuccess(result))
    } catch (error) {
      dispatch(hasErrored(error.message))
    }
  }
}