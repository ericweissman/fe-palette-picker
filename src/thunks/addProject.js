import { isLoading, hasErrored, addProjectSuccess } from '../actions'

export const addProject = (url, project_name) => {
  return async (dispatch) => {
    try {
      dispatch(isLoading(true))
      const response = await fetch(url, {
        method: "POST",
        headers: {
          "Content-Type" : "application/json"
        },
        body: JSON.stringify({ project_name })
      })
      if (!response.ok) {
        throw Error(response.statusText)
      }
      dispatch(isLoading(false))
      const result = await response.json()
      dispatch(addProjectSuccess(result))
    } catch (error) {
      dispatch(hasErrored(error.message))
    }
  }
}