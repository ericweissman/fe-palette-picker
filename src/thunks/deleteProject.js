import { isLoading, hasErrored, deleteProjectSuccess } from '../actions'

export const deleteProject = (url, projectID) => {
  return async (dispatch) => {
    try {
      dispatch(isLoading(true))
      const response = await fetch(url, {
        method: "DELETE",
        headers: {
          "Content-Type" : "application/json"
        }
      })
      if (!response.ok) {
        throw Error(response.statusText)
      }
      dispatch(isLoading(false))
      // const result = await response.json()
      dispatch(deleteProjectSuccess(projectID))
    } catch (error) {
      dispatch(hasErrored(error.message))
    }
  }
}