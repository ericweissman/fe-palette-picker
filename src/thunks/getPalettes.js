import { isLoading, hasErrored, getPalettesSuccess } from '../actions'

export const getPalettes = (url, id) => {
  return async (dispatch) => {
    try {
      dispatch(isLoading(true))
      const response = await fetch(url)
      if (!response.ok) {
        throw Error(response.statusText)
      }
      dispatch(isLoading(false))
      const result = await response.json()
      dispatch(getPalettesSuccess(result, id))
    } catch (error) {
      dispatch(hasErrored(error.message))
    }
  }
}