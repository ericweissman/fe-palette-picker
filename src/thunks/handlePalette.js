import { isLoading, hasErrored } from '../actions'

export const handlePalette = (url, actionToDispatch, method, palette) => {
  return async (dispatch) => {
    try {
      dispatch(isLoading(true))
      const response = await fetch(url, {
        method: `${method}`,
        headers: {
          "Content-Type": "application/json"
        },
        body: JSON.stringify(palette)
      })
      if (!response.ok) {
        throw Error(response.statusText)
      }
      dispatch(isLoading(false))
      const result = await response.json()
      dispatch(actionToDispatch(result))
    } catch (error) {
      dispatch(hasErrored(error.message))
    }
  }
}