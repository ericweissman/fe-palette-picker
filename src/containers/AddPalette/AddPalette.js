import React, { Component } from 'react'
import { connect } from 'react-redux'
import { getProjects } from '../../thunks/getProjects'
import { addPalette } from '../../thunks/addPalette'
import PropTypes from 'prop-types'

class AddPalette extends Component {
  state = {
    paletteName: '',
    projectID: -1,
  }

  handleChange = (e) => {
    const { name, value } = e.target
    this.setState({ [name]: value })
  }

  handleSubmit = (e) => {
    e.preventDefault();
    const { projectID, paletteName } = this.state
    const { activePalette } = this.props
    const url = process.env.REACT_APP_BACKEND_URL + `/api/v1/projects/${projectID}/palettes`
    const newPalette = {
      palette_name: paletteName,
      color_1: activePalette[0],
      color_2: activePalette[1],
      color_3: activePalette[2],
      color_4: activePalette[3],
      color_5: activePalette[4],
    }
    this.props.addPalette(url, newPalette)
  }

  render() {
    const { projects } = this.props
    const projectList = projects.map(project => {
      return (
        <option key={project.id} value={project.id}>
          {project.project_name}
        </option>
      )
    })

    return (
      <div className="add-palette">
        <h3>New palette</h3>
        <form>
          <input
            onChange={this.handleChange}
            placeholder='new palette name'
            name='paletteName'
            value={this.state.paletteName}>
          </input>
          <button onClick={this.handleSubmit}>
            Save to...
          </button>
          <select name="projectID" onChange={this.handleChange}>
            <option  hidden> select a project </option>
            {projects && projectList}
          </select>
        </form>
      </div>
    )
  }
}

export const mapStateToProps = (state) => ({
  projects: state.projects,
  activePalette: state.activePalette,
})

export const mapDispatchToProps = (dispatch) => ({
  getProjects: (url) => dispatch(getProjects(url)),
  addPalette: (url, palette) => dispatch(addPalette(url, palette)),
})

export default connect(mapStateToProps, mapDispatchToProps)(AddPalette)