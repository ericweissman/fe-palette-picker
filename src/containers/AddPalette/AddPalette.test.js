import React from 'react';
import { shallow } from 'enzyme';
import { handlePalette } from '../../thunks/handlePalette'
import { AddPalette, mapStateToProps, mapDispatchToProps } from './AddPalette'
import { addPaletteSuccess } from '../../actions'

jest.mock('../../thunks/handlePalette')

describe('AddPalette', () => {
  let wrapper

  describe('snapshots and initial state', () => {

    it('should have an initial state', () => {
      const mockProjects = []
      const mockActivePalette = []
      const expected = { paletteName: '', projectID: -1, errorMessage: '' }
      wrapper = shallow(
        <AddPalette projects={mockProjects} activePalette={mockActivePalette} />
      )
      
      expect(wrapper.state()).toEqual(expected)
    })

    it('should match the initial snapshot', () => {
      const mockProjects = []
      const mockActivePalette = []

      wrapper = shallow(
        <AddPalette projects={mockProjects} activePalette={mockActivePalette} />
      )

      expect(wrapper).toMatchSnapshot()
    })

    it('should match the snapshot with projects and activePalette', () => {
      const mockProjects = [{ id: 1, project_name: 'first' }]
      const mockActivePalette = ['111111', '222222', '333333', '444444', '555555']

      wrapper = shallow(
        <AddPalette projects={mockProjects} activePalette={mockActivePalette} />
      )

      expect(wrapper).toMatchSnapshot()
    })
  })

  describe('mapStateToProps', () => {
    it('should return an object with a projects array and activePalette array', () => {
      const mockState = { projects: [{}, {}], activePalette: ['1', '2', '3', '4', '5'], palettes: [], paletteToEdit: {} }
      const expected = { projects: [{}, {}], activePalette: ['1', '2', '3', '4', '5'] }

      const mappedProps = mapStateToProps(mockState)

      expect(mappedProps).toEqual(expected)
    })
  })

  describe('mapDispatchToProps', () => {
    it('should call dispatch', () => {
      const mockDispatch = jest.fn()
      const mockURL = 'http://localhost:3001/api/v1/projects/1/palettes'
      const mockPalette = {}

      const actionToDispatch = handlePalette(mockURL, addPaletteSuccess, 'POST', mockPalette)

      const mappedProps = mapDispatchToProps(mockDispatch)
      mappedProps.handlePalette(mockURL, addPaletteSuccess, 'POST', mockPalette)

      expect(mockDispatch).toHaveBeenCalledWith(actionToDispatch)
    })
  })
})