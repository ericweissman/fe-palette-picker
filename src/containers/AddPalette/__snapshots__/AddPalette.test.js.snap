// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddPalette snapshots and initial state should match the initial snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <AddPalette
    activePalette={Array []}
    projects={Array []}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div>
          <h3>
            New palette
          </h3>
          <h5>
            
          </h5>
        </div>,
        <form>
          <input
            name="paletteName"
            onChange={[Function]}
            placeholder="new palette name"
            value=""
          />
          <button
            onClick={[Function]}
          >
            Save to...
          </button>
          <select
            name="projectID"
            onChange={[Function]}
          >
            <option
              hidden={true}
            >
               select a project 
            </option>
          </select>
        </form>,
      ],
      "className": "add-palette",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <h3>
              New palette
            </h3>,
            <h5>
              
            </h5>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "New palette",
            },
            "ref": null,
            "rendered": "New palette",
            "type": "h3",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "",
            },
            "ref": null,
            "rendered": "",
            "type": "h5",
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <input
              name="paletteName"
              onChange={[Function]}
              placeholder="new palette name"
              value=""
            />,
            <button
              onClick={[Function]}
            >
              Save to...
            </button>,
            <select
              name="projectID"
              onChange={[Function]}
            >
              <option
                hidden={true}
              >
                 select a project 
              </option>
            </select>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "name": "paletteName",
              "onChange": [Function],
              "placeholder": "new palette name",
              "value": "",
            },
            "ref": null,
            "rendered": null,
            "type": "input",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Save to...",
              "onClick": [Function],
            },
            "ref": null,
            "rendered": "Save to...",
            "type": "button",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <option
                  hidden={true}
                >
                   select a project 
                </option>,
                Array [],
              ],
              "name": "projectID",
              "onChange": [Function],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": " select a project ",
                  "hidden": true,
                },
                "ref": null,
                "rendered": " select a project ",
                "type": "option",
              },
            ],
            "type": "select",
          },
        ],
        "type": "form",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div>
            <h3>
              New palette
            </h3>
            <h5>
              
            </h5>
          </div>,
          <form>
            <input
              name="paletteName"
              onChange={[Function]}
              placeholder="new palette name"
              value=""
            />
            <button
              onClick={[Function]}
            >
              Save to...
            </button>
            <select
              name="projectID"
              onChange={[Function]}
            >
              <option
                hidden={true}
              >
                 select a project 
              </option>
            </select>
          </form>,
        ],
        "className": "add-palette",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <h3>
                New palette
              </h3>,
              <h5>
                
              </h5>,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "New palette",
              },
              "ref": null,
              "rendered": "New palette",
              "type": "h3",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "",
              },
              "ref": null,
              "rendered": "",
              "type": "h5",
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <input
                name="paletteName"
                onChange={[Function]}
                placeholder="new palette name"
                value=""
              />,
              <button
                onClick={[Function]}
              >
                Save to...
              </button>,
              <select
                name="projectID"
                onChange={[Function]}
              >
                <option
                  hidden={true}
                >
                   select a project 
                </option>
              </select>,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "name": "paletteName",
                "onChange": [Function],
                "placeholder": "new palette name",
                "value": "",
              },
              "ref": null,
              "rendered": null,
              "type": "input",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Save to...",
                "onClick": [Function],
              },
              "ref": null,
              "rendered": "Save to...",
              "type": "button",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <option
                    hidden={true}
                  >
                     select a project 
                  </option>,
                  Array [],
                ],
                "name": "projectID",
                "onChange": [Function],
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": " select a project ",
                    "hidden": true,
                  },
                  "ref": null,
                  "rendered": " select a project ",
                  "type": "option",
                },
              ],
              "type": "select",
            },
          ],
          "type": "form",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
  Symbol(enzyme.__childContext__): null,
}
`;

exports[`AddPalette snapshots and initial state should match the snapshot with projects and activePalette 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <AddPalette
    activePalette={
      Array [
        "111111",
        "222222",
        "333333",
        "444444",
        "555555",
      ]
    }
    projects={
      Array [
        Object {
          "id": 1,
          "project_name": "first",
        },
      ]
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div>
          <h3>
            New palette
          </h3>
          <h5>
            
          </h5>
        </div>,
        <form>
          <input
            name="paletteName"
            onChange={[Function]}
            placeholder="new palette name"
            value=""
          />
          <button
            onClick={[Function]}
          >
            Save to...
          </button>
          <select
            name="projectID"
            onChange={[Function]}
          >
            <option
              hidden={true}
            >
               select a project 
            </option>
            <option
              value={1}
            >
              first
            </option>
          </select>
        </form>,
      ],
      "className": "add-palette",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <h3>
              New palette
            </h3>,
            <h5>
              
            </h5>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "New palette",
            },
            "ref": null,
            "rendered": "New palette",
            "type": "h3",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "",
            },
            "ref": null,
            "rendered": "",
            "type": "h5",
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <input
              name="paletteName"
              onChange={[Function]}
              placeholder="new palette name"
              value=""
            />,
            <button
              onClick={[Function]}
            >
              Save to...
            </button>,
            <select
              name="projectID"
              onChange={[Function]}
            >
              <option
                hidden={true}
              >
                 select a project 
              </option>
              <option
                value={1}
              >
                first
              </option>
            </select>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "name": "paletteName",
              "onChange": [Function],
              "placeholder": "new palette name",
              "value": "",
            },
            "ref": null,
            "rendered": null,
            "type": "input",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Save to...",
              "onClick": [Function],
            },
            "ref": null,
            "rendered": "Save to...",
            "type": "button",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <option
                  hidden={true}
                >
                   select a project 
                </option>,
                Array [
                  <option
                    value={1}
                  >
                    first
                  </option>,
                ],
              ],
              "name": "projectID",
              "onChange": [Function],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": " select a project ",
                  "hidden": true,
                },
                "ref": null,
                "rendered": " select a project ",
                "type": "option",
              },
              Object {
                "instance": null,
                "key": "1",
                "nodeType": "host",
                "props": Object {
                  "children": "first",
                  "value": 1,
                },
                "ref": null,
                "rendered": "first",
                "type": "option",
              },
            ],
            "type": "select",
          },
        ],
        "type": "form",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div>
            <h3>
              New palette
            </h3>
            <h5>
              
            </h5>
          </div>,
          <form>
            <input
              name="paletteName"
              onChange={[Function]}
              placeholder="new palette name"
              value=""
            />
            <button
              onClick={[Function]}
            >
              Save to...
            </button>
            <select
              name="projectID"
              onChange={[Function]}
            >
              <option
                hidden={true}
              >
                 select a project 
              </option>
              <option
                value={1}
              >
                first
              </option>
            </select>
          </form>,
        ],
        "className": "add-palette",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <h3>
                New palette
              </h3>,
              <h5>
                
              </h5>,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "New palette",
              },
              "ref": null,
              "rendered": "New palette",
              "type": "h3",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "",
              },
              "ref": null,
              "rendered": "",
              "type": "h5",
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <input
                name="paletteName"
                onChange={[Function]}
                placeholder="new palette name"
                value=""
              />,
              <button
                onClick={[Function]}
              >
                Save to...
              </button>,
              <select
                name="projectID"
                onChange={[Function]}
              >
                <option
                  hidden={true}
                >
                   select a project 
                </option>
                <option
                  value={1}
                >
                  first
                </option>
              </select>,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "name": "paletteName",
                "onChange": [Function],
                "placeholder": "new palette name",
                "value": "",
              },
              "ref": null,
              "rendered": null,
              "type": "input",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Save to...",
                "onClick": [Function],
              },
              "ref": null,
              "rendered": "Save to...",
              "type": "button",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <option
                    hidden={true}
                  >
                     select a project 
                  </option>,
                  Array [
                    <option
                      value={1}
                    >
                      first
                    </option>,
                  ],
                ],
                "name": "projectID",
                "onChange": [Function],
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": " select a project ",
                    "hidden": true,
                  },
                  "ref": null,
                  "rendered": " select a project ",
                  "type": "option",
                },
                Object {
                  "instance": null,
                  "key": "1",
                  "nodeType": "host",
                  "props": Object {
                    "children": "first",
                    "value": 1,
                  },
                  "ref": null,
                  "rendered": "first",
                  "type": "option",
                },
              ],
              "type": "select",
            },
          ],
          "type": "form",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
  Symbol(enzyme.__childContext__): null,
}
`;
